metron-platform/metron-test-utilities/src/main/java/org/apache/metron/test/mock/MockHTable.java

```diff
@@ -70,7 +93,7 @@ public static void clear() {
   private final String tableName;
   private final List<String> columnFamilies = new ArrayList<>();
   private HColumnDescriptor[] descriptors;
-
+  private List<Put> putLog;
```

Comment:
That's what I initially tried and it was not working.  Thinking back, I later found two other issues and had to fix those also (described above as MockHTable scanner and Test Validation.)  So the problem may not have been caused by ImmtableList.copy + synchronizedList, but by those other issues.  Dunno.